### ISSUE TITLE
Remove incorrect HTML markup from Clarity stack view

### ISSUE THREAD

**Issue by bashibozuk** [vmware/clarity#3984]
## Summary

&lt;dt&gt; and &lt;dd&gt; tags used as a markup in the stack view should be placed only as children of &lt;dl&gt; tag. When mark up is not valid, browsers and screen readers will not interpret the structure of content correctly.

* What is the change?
remove &lt;dt&gt; and &lt;dd&gt; tags from the stack view 
* Why should it go in Clarity?
Provide valid HTML markup and better accessibility
* Does this change impact existing behaviors? If so how?
No
* If this change introduces a new behavior, is this behavior accessible?
Yes
## Use case

As a screen reader user of the Clarity Stack view I want its markup to be interpreted correctly

## Examples

N/A
## Workarounds

N/A

**Comment by gnomeontherun** [comment]
@bashibozuk Is this based on automatic testing or manual testing? @chlane has done extensive testing of Clarity components and did not report an issue about this.

**Comment by chlane** [comment]
Yes there is an HTML validity problem here. There is a dl tag but there should not be any div elements elements wrapping the dt and dd elements. 

<dl _ngcontent-clarity-c1="" class="stack-view"><div _ngcontent-clarity-c1="" class="stack-block"><dt _ngcontent-clarity-c1="" class="stack-block-label">Label 1</dt><dd _ngcontent-clarity-c1="" class="stack-block-content">Content 1</dd></div><div _ngcontent-clarity-c1="" class="stack-block stack-block-expandable stack-block-expanded"><dt _ngcontent-clarity-c1="" class="stack-block-label">Label 2</dt><dd _ngcontent-clarity-c1="" class="stack-block-content">Content 2</dd><div _ngcontent-clarity-c1="" class="stack-children"><div _ngcontent-clarity-c1="" class="stack-block"><dt _ngcontent-clarity-c1="" class="stack-block-label">Sub-label 1</dt><dd _ngcontent-clarity-c1="" class="stack-block-content">Sub-content 1</dd></div><div _ngcontent-clarity-c1="" class="stack-block"><dt _ngcontent-clarity-c1="" class="stack-block-label">Sub-label 2</dt><dd _ngcontent-clarity-c1="" class="stack-block-content">Sub-content 2</dd></div><div _ngcontent-clarity-c1="" class="stack-block"><dt _ngcontent-clarity-c1="" class="stack-block-label">Sub-label 3</dt><dd _ngcontent-clarity-c1="" class="stack-block-content">Sub-content 3</dd></div></div></div><div _ngcontent-clarity-c1="" class="stack-block stack-block-expandable"><dt _ngcontent-clarity-c1="" class="stack-block-label">Label 3</dt><dd _ngcontent-clarity-c1="" class="stack-block-content">Content 3</dd><div _ngcontent-clarity-c1="" class="stack-children"></div></div></dl>

**Comment by gnomeontherun** [comment]
I understand there may be a validity problem, but is there a real impact to user? We couldn't change this without a full rewrite and potentially dropping some features, so I have to ask if this is broken for users or just a static analysis tool violation?

**Comment by chlane** [comment]
NVDA/Firefox and VoiceOver/Safari users will not hear that they are reading a definition list, the size of the list or the position of the item they are reading within the list as they would with a properly marked up example as seen at http://cx-guidelines.vmware.com/examples/dlList.html. Sheri says we can fix it in a future version of Clarity if we can't do it now.

**Comment by mathisscott** [comment]
@gnomeontherun : Was this fixed by #4142 ?

**Comment by gnomeontherun** [comment]
Yes in v3+, the `dd` and `dt` structure is now replaced with divs and has proper aria labels. Since it is a breaking change, it can't be backported.

Closing as resolved.
